internal/handlers/pg/msg_find.go

```diff
@@ -154,3 +169,67 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 
 	return &reply, nil
 }
+
+// getMaxTimeMSParameter gets maxTimeMS optional parameter.
+func getMaxTimeMSParameter(document *types.Document) (int32, error) {
+	var maxTimeMS int32
+	maxTimeMSParam, err := common.GetOptionalParam(document, "maxTimeMS", maxTimeMS)
+	if err != nil {
+		var commonErr *common.Error
+		if errors.As(err, &commonErr) {
+			return 0, getErrorForInvalidMaxTimeMS(document)
+		}
+
+		return 0, err
+	}
+
+	if maxTimeMSParam < 0 {
+		return 0, common.NewErrorMsg(
+			common.ErrBadValue,
+			fmt.Sprintf("%v value for maxTimeMS is out of range", maxTimeMSParam),
+		)
+	}
+
+	return maxTimeMSParam, nil
+}
+
+// getErrorForInvalidMaxTimeMS returns error for invalid maxTimeMS type.
+func getErrorForInvalidMaxTimeMS(document *types.Document) error {
+	v, err := document.Get("maxTimeMS")
+	if err != nil {
+		return nil
+	}
```

Comment:
if you don't expect error here, you can replace with it with this:
```suggestion
	v := must.NotFail(document.Get("maxTimeMS"))
```

Reply:
mus.NotFail will panic.
previous solution - not.

Reply:
Fixed